### ISSUE TITLE
Question about loss divergence when training AKOrN on custom datasets

### ISSUE THREAD

**Issue by ugadiarov-la-phystech-edu** [autonomousvision/akorn#6]
Hello! Thank you for releasing the code for your project — it's been very helpful.
In my experiments, I use AKOrN as a model for object discovery on custom datasets. I’ve noticed that on some datasets, the loss starts to increase after a few epochs and eventually leads to training divergence. This usually happens when the images contain small objects with distinct colors. I’ve tried using learning rate decay to prevent divergence. While this helps somewhat, the loss still seems to converge prematurely to a relatively high value, suggesting there’s room for improvement. Interestingly, when I use datasets from the same domain but with images that contain larger objects but with the same colors, the loss steadily decreases to low values during training.

Have you encountered similar issues with loss divergence in your research? If so, could you share how you addressed them?

Thanks again for your work!

**Comment by takerum** [comment]
Hi, thank you for the question! I've encountered such divergence or NaN when I set a too high learning rate at the initialization, but never seen the loss explode in the middle of training. What dataset and training method do you use for your case?

**Comment by takerum** [comment]
Thanks for sharing!
Other than lowering the learning rate, you could try increasing the oscillator dimensions `N' or lowering the step size of the Kuramoto step `gamma'. We observe that setting N=2 leads to overly strong regularity and causes underfitting. 
Regarding the oscillator dimension, we also observe a slight improvement with larger `N' (see the fig 16 of the paper).
